assignment3.md

```diff
@@ -0,0 +1,233 @@
+
+```
+python evaluate.py <GROUND-TRUTH-FILE> <PREDICTED-CLUSTERS-FILE>
+```
+
+Your Tasks
+======================
+You have 3 tasks. 
+
+### 1. Sparse Representations 
+
+Your first task is to generate clusters for the target words in `test_vocab.txt` based on a feature-based (not dense) vector space representation. In this type of VSM, each dimension of the vector space corresponds to a specific feature, such as a context word (see, for example, the term-context matrix described in [Chapter 15.1.2 of Jurafsky & Martin](https://web.stanford.edu/~jurafsky/slp3/15.pdf)). 
+
+For this task, use files:
+* `makecooccurrences.py`: to calculate cooccurrence statistics and write vectors to file (this could take a long time (>1 hour) to make plans accordingly)
+* `question1.py`: to load cooccurrences vectors and cluster.
+
+We have provided a corpus for feature extraction at `/home1/a/acocos/data/reuters.rcv1.tokenized.gz`, and also code to read it. 
+
+Here is an example of the K-means code:
+
+{% highlight python %}
+from sklearn.cluster import KMeans
+kmeans = KMeans(n_clusters=k).fit(X)
+kmeans.labels_
+{% endhighlight %}
+
+
+<!--- Baseline description is a placeholder --->
+The baseline system for this section represents words using a term-context matrix `M` of size `|V| x D`, where `|V|` is the size of the vocabulary and D=500. Each feature corresponds to one of the top 500 most-frequent words in the corpus. The value of matrix entry `M[i][j]` gives the number of times the context word represented by column `j` appeared within W=3 words to the left or right of the word represented by row `i` in the corpus. Using this representation, the baseline system clusters each paraphrase set using K-means.  
+
+Implementing the baseline will score you a B, but why not try and see if you can do better? You might try experimenting with different features, for example:
+- What if you reduce or increase `D` in the baseline implementation?
+- Does it help to change the window `W` used to extract contexts?
+- Play around with the feature weighting -- instead of raw counts, would it help to use PMI or TF-IDF?
+- Try a different clustering algorithm that's included with the [scikit-learn clustering package](http://scikit-learn.org/stable/modules/clustering.html), or implement your own.
+- What if you include additional types of features, like paraphrases in the [Paraphrase Database](http://www.paraphrase.org) or the part-of-speech of context words?
+
+The only feature types that are off-limits are WordNet features.
+
+Turn in the predicted clusters that your VSM generates. Also provide a brief description of your method in a file called `writeup.pdf`, making sure to describe the vector space model you chose, the clustering algorithm you used, and the results of any preliminary experiments you might have run on the training set.
+
+The output file should be called: `test_clusters_features.txt`
+
+### 2. Questions
```

Comment:
I am going to move the detailed questions themselves from `question2.txt` to this space, and leave just the blanks in `question2.txt`.